src/api/firebase.js

```diff
@@ -93,13 +96,31 @@ export async function updateItem(listId, itemData) {
 	 * this function must accept!
 	 */
 	const itemRef = doc(db, listId, itemData.id);
+	console.log('in updateItem');
+	let dayDiff;
+	if (itemData.dateLastPurchased) {
+		dayDiff = getDaysBetweenDates(itemData.dateLastPurchased);
+	} else {
+		dayDiff = getDaysBetweenDates(itemData.dateCreated);
+	}
+
+	let previousEstimate = parseInt(itemData.previousEstimate);
```

Comment:
This parseInt is interesting & I'm not very familiar with what this does. Would love to hear more during demos about why this is being used and what is happening here.

Reply:
The value from the input when an item is added to the list is stored as a string, we are converting it to a number to re-calculate the time frame

Reply:
I found this explanation really helpful for examples on the parsing https://www.w3schools.com/jsref/jsref_parseint.asp

I know the parameters are a string and a radix, MDN docs says you can't always rely that it will default to base 10 so more safe syntax is parseInt(string, 10) <-- adding the radix as the 2nd parameter, in this case base 10.

I wasn't super familiar with radix but I like this explanation https://www.computerhope.com/jargon/r/radix.htm#:~:text=Radix%20is%20a%20term%20used,therefore%2C%20its%20radix%20is%2010.

![radix](https://user-images.githubusercontent.com/89806097/183269717-1b0cbf4e-1f8a-497a-9154-be285583094b.PNG)